pkg/resource/firewall/hooks.go

```diff
@@ -88,13 +134,57 @@ func (rm *resourceManager) syncFirewallPolicyARN(
 	return nil
 }
 
+// addPutFieldsToSpec will describe logging config and add its
+// returned values to the Firewall spec.
+func (rm *resourceManager) addLoggingConfigToSpec(
+	ctx context.Context,
+	r *resource,
+	ko *svcapitypes.Firewall,
+) (err error) {
+	getLoggingConfigurationResponse, err := rm.sdkapi.DescribeLoggingConfigurationWithContext(ctx, rm.getLoggingConfigurationPayload(r))
+	if err != nil {
+		return err
+	}
+	ko.Spec.LoggingConfiguration = rm.setResourceLoggingConfiguration(r, getLoggingConfigurationResponse)
+	return nil
+}
+
 func customPreCompare(
 	delta *ackcompare.Delta,
 	a *resource,
 	b *resource,
 ) {
 	// Custom comparison of subnet mappings such that they can be compared in a deterministic way.
 	customCompareSubnetMappings(delta, a, b)
+
+	// Custom comparison of logging configurations such that they can be
+	// compared in a deterministic way.
+	customCompareLoggingConfigurations(delta, a, b)
+}
+
+func customCompareLoggingConfigurations(
+	delta *ackcompare.Delta,
+	a *resource,
+	b *resource,
+) {
+	if a.ko.Spec.LoggingConfiguration == nil && b.ko.Spec.LoggingConfiguration == nil {
+		return
+	}
+
+	if a.ko.Spec.LoggingConfiguration == nil {
+		a.ko.Spec.LoggingConfiguration = &svcapitypes.LoggingConfiguration{}
+	}
+
+	if b.ko.Spec.LoggingConfiguration == nil {
+		b.ko.Spec.LoggingConfiguration = &svcapitypes.LoggingConfiguration{}
+	}
+
+	sort.Slice(a.ko.Spec.LoggingConfiguration.LogDestinationConfigs[:], func(i, j int) bool {
+		return *a.ko.Spec.LoggingConfiguration.LogDestinationConfigs[i].LogType < *a.ko.Spec.LoggingConfiguration.LogDestinationConfigs[j].LogType
+	})
+	sort.Slice(b.ko.Spec.LoggingConfiguration.LogDestinationConfigs[:], func(i, j int) bool {
+		return *b.ko.Spec.LoggingConfiguration.LogDestinationConfigs[i].LogType < *b.ko.Spec.LoggingConfiguration.LogDestinationConfigs[j].LogType
```

Comment:
Can we make a copy of `a` and `b` before sorting their elements? the controller patches back the desired state (a) which can cause drift and unwanted behaviour if you're using ACK with a gitops tool

Reply:
Done.
Along with this, I also had to undo initialization of `a.ko.Spec.LoggingConfiguration` , which then needed additional null check changes at many places.